google-cloud-dialogflow/MIGRATING.md

```diff
@@ -0,0 +1,274 @@
+    over endpoint address, network timeouts, and retry. See
+    [Client Configuration](#client-configuration) for more info. Furthermore,
+    when creating a client object, you can customize its configuration in a
+    block rather than passing arguments to the constructor. See
+    [Creating Clients](#creating-clients) for more info.
+ *  Previously, methods typically had at least one positional argument. Now,
+    all arguments are keyword arguments. Additionally, you can pass a proto
+    request object instead of separate arguments. See
+    [Passing Arguments](#passing-arguments) for more info.
+ *  Previously, some client classes included class methods for constructing
+    resource paths. These paths are now instance methods, and are also
+    available in a separate paths module. See
+    [Resource Path Helpers](#resource-path-helpers) for more info.
+ *  Some classes have moved into different namespaces. See
+    [Class Namespaces](#class-namespaces) for more info.
+
+### Library Structure
+
+Older 0.x releases of the `google-cloud-dialogflow` gem were all-in-one gems that
+included potentially multiple clients for multiple versions of the Dialogflow
+service. Factory methods such as `Google::Cloud::Dialogflow::Agents.new()` would
+return you client instances such as `Google::Cloud::Dialogflow::V2::AgentsClient`.
+These classes were all defined in the same gem.
+
+With the 1.0 release, the `google-cloud-dialogflow` gem still provides factory
+methods for obtaining clients. (The method signatures will have changed. See
+[Creating Clients](#creating-clients) for details.) However, the actual client
+classes have been moved into separate gems, on per service version. Currently,
+Dialogflow has one version, V2. The
+`Google::Cloud::Dialogflow::V2::Agents::Client` class, along with its
+helpers and data types, are now part of the `google-cloud-dialogflow-v2` gem.
+
+For normal usage, you can continue to install the `google-cloud-dialogflow` gem
+and continue to use factory methods to create clients. This will remain the
+easiest way to use the Dialogflow client. However, you may alternatively
+choose to install only one of the versioned gems. For example, if you know you
+will only use `V2` of the service, you can install `google-cloud-dialogflow-v2`
+by itself, and construct instances of the
+`Google::Cloud::Dialogflow::V2::Agents::Client` client class directly.
+
+### Client Configuration
+
+In older releases, if you wanted to customize performance parameters or
+low-level behavior of the client (such as credentials, timeouts, or
+instrumentation), you would pass a variety of keyword arguments to the client
+constructor. It was also extremely difficult to customize the default settings.
+
+With the 1.0 release, a configuration interface provides access to these
+parameters, both global defaults and per-client settings. For example, to set
+global credentials and default timeout for all Dialogflow V2 sessions clients:
```

Comment:
Any way to avoid the use of "global" here? Since the broadest scope is "all Dialogflow versions and services" (discussed below), this usage of "global" seems inconsistent.

Reply:
Done